src/sentry/receivers/rule_snooze.py

```diff
@@ -1,45 +1,46 @@
-from django.db.models.signals import post_save, pre_delete
+from django.contrib.auth.models import AnonymousUser
 
+from sentry import audit_log
 from sentry.models.rulesnooze import RuleSnooze
+from sentry.users.models.user import User
+from sentry.users.services.user import RpcUser
+from sentry.utils.audit import create_audit_entry_from_user
 from sentry.workflow_engine.models import AlertRuleDetector, AlertRuleWorkflow
 
 
-def _update_workflow_engine_models(instance: RuleSnooze, is_enabled: bool) -> None:
+def _update_workflow_engine_models(
+    instance: RuleSnooze, is_enabled: bool, requesting_user: User | AnonymousUser
+) -> None:
     if instance.user_id is not None:
         return
 
     if instance.rule:
         alert_rule_workflow = AlertRuleWorkflow.objects.filter(rule_id=instance.rule.id).first()
         if alert_rule_workflow and alert_rule_workflow.workflow:
-            alert_rule_workflow.workflow.update(enabled=is_enabled)
-
+            workflow = alert_rule_workflow.workflow
+            workflow.update(enabled=is_enabled)
+            if isinstance(requesting_user, User) or isinstance(requesting_user, RpcUser):
```

Comment:
Typing got all garbled here - from the actual request it's a `User` or `AnonymousUser` but when we run tests it's an `RpcUser`. `create_audit_entry_from_user` takes either `User` or `RpcUser` so I had to do some annoying massaging - if there's a way to force the test to pass `User` that'd be better but I'm not sure how to do that.